api/request.go

```diff
@@ -53,20 +90,46 @@ type Normalizable interface {
 }
 
 func (r *RequestedResource) NormalizeConnection() error {
-	url, err := url.Parse(r.Vds)
-	if err != nil {
-		return core.NewInvalidArgument(err.Error())
+
+	if len(r.Vds) == 0 || r.Vds[0] == "" {
+		return core.NewInvalidArgument("No VDS url provided")
+	}
+
+	signedUrls := false
+	if len(r.Sas) == 0 || r.Sas[0] == "" {
+		// All vds urls are signed
+		r.Sas = []string{}
```

Comment:
```
vds1?sas1, vds2?sas2
"", "hahaha"
```

```
(no binary operator)
vds: vds1?sas1
sas: "", "haha"
```

```

vds: [vds1?sas1, vds2?sas2]
sas: [""]
```


Should any of these fail? Because they do not now.